### PR TITLE
Allow files paths for include URIs

### PR DESCRIPTION
**Pull request by azeey**
`//include/uri` used to only take directories that contain a model.config file and the sdformat file. However, allowing `//include/uri` to take file names can be supported without loss of functionality and will allow users to specify non-sdformat files to include via the [Interface API](http://sdformat.org/tutorials?tut=composition_proposal&cat=pose_semantics_docs&#1-5-minimal-libsdformat-interface-types-for-non-sdformat-models), when it's implemented.

Toward #430

### CODE REVIEW

File: test/integration/includes.cc [hunk 1]
```diff
@@ -209,6 +209,19 @@ TEST(IncludesTest, Includes)
...
+  EXPECT_EQ("test_model_with_file", model2->Name());
+  EXPECT_FALSE(model2->Static());
+  EXPECT_EQ(1u, model2->LinkCount());
+  ASSERT_FALSE(nullptr == model2->LinkByIndex(0));
+  ASSERT_FALSE(nullptr == model2->LinkByName("link"));
+  EXPECT_EQ(model2->LinkByName("link")->Name(), model2->LinkByIndex(0)->Name());
+  EXPECT_TRUE(nullptr == model2->LinkByIndex(1));
+  EXPECT_TRUE(model2->LinkNameExists("link"));
+  EXPECT_FALSE(model2->LinkNameExists("coconut"));
+  EXPECT_EQ("1.6", model2->Element()->OriginalVersion());
```
Comment (brawner) [inline test/integration/includes.cc:224]: Does the version number of the includes.sdf need to be bumped to 1.8?

File: test/integration/includes.cc [hunk 2]
```diff
@@ -209,6 +209,19 @@ TEST(IncludesTest, Includes)
   EXPECT_EQ("", model1->PoseRelativeTo());
   ASSERT_NE(nullptr, model1->Element());
   EXPECT_TRUE(model1->Element()->HasElement("plugin"));
+
+  const sdf::Model *model2 = world->ModelByIndex(2);
+  ASSERT_NE(nullptr, model2);
+  EXPECT_EQ("test_model_with_file", model2->Name());
+  EXPECT_FALSE(model2->Static());
+  EXPECT_EQ(1u, model2->LinkCount());
+  ASSERT_FALSE(nullptr == model2->LinkByIndex(0));
```
Comment (brawner) [inline test/integration/includes.cc:218]: `ASSERT_NE`? Here and lines 219 and 221

**Review by brawner — APPROVED** [review]
Looks good to me with some minor nitpicks.

File: test/integration/includes.cc (hunk 2 above)
Comment (azeey) [reply test/integration/includes.cc]: Fixed in 5b80879

File: test/integration/includes.cc (hunk 1 above)
Comment (azeey) [reply test/integration/includes.cc:224]: Bumped to 1.8 in 5b80879.

**Review by EricCousineau-TRI — APPROVED** [review]
Sweet, looks great!

**Review by scpeters — APPROVED** [review]

**Pull request merged**